src/blueapi/worker/task.py

```diff
@@ -41,15 +41,18 @@ class RunPlan(Task):
     def do_task(self, ctx: BlueskyContext) -> None:
         LOGGER.info(f"Asked to run plan {self.name} with {self.params}")
 
-        plan = ctx.plans[self.name]
-        func = ctx.plan_functions[self.name]
-        sanitized_params = _lookup_params(ctx, plan, self.params)
-        plan_generator = func(**sanitized_params.dict())
+        metadata = ctx.find_plan_metadata(self.name)
```

Comment:
If these two functions are only ever called together and both have to be present, it'd be clearer to have a single `find_plan(name) -> Optional[(Plan, PlanFunc)]`. Probably suggests `plans` and `plan_functions` should be merged into a single field in the context as well. Maybe a dict of `name -> (Plan, func)`?